### PR TITLE
Sprint b02   get user

### PR DESCRIPTION
**Pull request by Leonorjp**

### CODE REVIEW

File: src/main/java/com/criticalsoftware/Annoucement.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+public class Annoucement extends PanacheMongoEntity {
+
+    private Product product;
+    private LocalDateTime date;
+    private User userDonor;
+    private User userDonee;
+
+    public Annoucement() {
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Annoucement.java]: Can we try without the empty constructor?

File: src/main/java/com/criticalsoftware/Annoucement.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+
+    public Annoucement(Product product, LocalDateTime date, User userDonor, User userDonee) {
+        this.product = product;
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Annoucement.java:18]: no need for date to passed into the method.
You can just write: this.date = LocalDateTime.now()

It just saves the date at that instant.

File: src/main/java/com/criticalsoftware/Contact.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import io.quarkus.mongodb.panache.PanacheMongoEntity;
+
+public class Contact extends PanacheMongoEntity {
+
+    private String email;
+    private int phoneNumber;
+    private String address;
+
+    public Contact() {
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Contact.java]: Can we try without the empty constructor?

File: src/main/java/com/criticalsoftware/Contact.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    public Contact(String email, int phoneNumber, String address) {
+        this.email = email;
+        this.phoneNumber = phoneNumber;
+        this.address = address;
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Contact.java]: extra line.

File: src/main/java/com/criticalsoftware/Annoucement.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        this.date = date;
+        this.userDonor = userDonor;
+        this.userDonee = userDonee;
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Annoucement.java]: extra line.

File: src/main/java/com/criticalsoftware/User.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+@MongoEntity(collection = "users")
+public class User extends PanacheMongoEntity {
+
+    private String name;
+    private LocalDate dateBirth;
+    private Contact contact;
+
+    public User() {
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/User.java]: Can we try without the empty constructor?

File: src/main/java/com/criticalsoftware/User.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public User(String name, LocalDate dateBirth, Contact contact) {
+        this.name = name;
+        this.dateBirth = dateBirth;
+        this.contact = contact;
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/User.java]: extra space.

File: src/main/java/com/criticalsoftware/UserResource.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        public Response get() {
+            return Response.ok(repository.listAll()).build();
+        }
+
+        //his method handles GET requests to /users/search/{name}
+        //It retrieves a user by its name and returns it in the response.
+        //If no such user is found, it returns a 404 Not Found status.
+        @GET
+        @Path("/search/{name}")
+        public Response search(@PathParam("name") String name) {
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/UserResource.java]: There is a mismatch. 
The first method should change the name (getById maybe?)
The second: (getByName, maybe?)
One of them has search on it. Both should have it or none.

**Review by jpmoreira-training — CHANGES_REQUESTED** [review]

File: src/main/java/com/criticalsoftware/Product.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+public class Product {
+
+    private String description;
+    private BufferedImage photo;
+    private String category;
+    private User user;
+
+
+    public Product() {
+    }
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Product.java:14]: Can we try without the empty constructor?

File: src/main/java/com/criticalsoftware/Product.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+    public Product(String description, BufferedImage photo, String category, User user) {
+        this.description = description;
+        this.photo = photo;
+        this.category = category;
+        this.user = user;
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Product.java]: extra line

File: src/main/java/com/criticalsoftware/UserResource.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        public Response get(@PathParam("id") String id){
+            User user = repository.findById(new ObjectId(id));
+            return Response.ok(user).build();
+        }
+
+        //This method handles GET requests to /users.
+        //It retrieves all users entities and returns them in the response.
+        @GET
+        public Response get() {
+            return Response.ok(repository.listAll()).build();
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/UserResource.java]: getAll(). What do you think?

File: src/main/java/com/criticalsoftware/UserResource.java [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Inject
+    private UserRepository repository;
+
+        //This method handles GET requests to /users/{id}.
+        //It retrieves a user by its ID and returns it in the response.
+        @GET
+        @Path("/{id}")
+        public Response get(@PathParam("id") String id){
+            User user = repository.findById(new ObjectId(id));
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/UserResource.java:24]: UserResponse userReponse = new UserResponde(user.getId(), user.getName(), ....)

File: src/main/java/com/criticalsoftware/User.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
+package com.criticalsoftware;
+import io.quarkus.mongodb.panache.PanacheMongoEntity;
+import io.quarkus.mongodb.panache.common.MongoEntity;
+import java.time.LocalDate;
+
+@MongoEntity(collection = "users")
+public class User extends PanacheMongoEntity {
+
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/User.java:9]: private ObjectId id;

**Review by jpmoreira-training — APPROVED** [review]
.

**Review by jpmoreira-training — CHANGES_REQUESTED** [review]
.

File: src/main/java/com/criticalsoftware/Annoucement.java (hunk 2 above)
Comment (Leonorjp) [reply src/main/java/com/criticalsoftware/Annoucement.java:18]: Done!

File: src/main/java/com/criticalsoftware/Contact.java (hunk 3 above)
Comment (Leonorjp) [reply src/main/java/com/criticalsoftware/Contact.java]: Doesn't work without

File: src/main/java/com/criticalsoftware/User.java (hunk 6 above)
Comment (Leonorjp) [reply src/main/java/com/criticalsoftware/User.java]: Doesn't work without.

File: src/main/java/com/criticalsoftware/Product.java (hunk 9 above)
Comment (Leonorjp) [reply src/main/java/com/criticalsoftware/Product.java:14]: Doesn't work without.

File: src/main/java/com/criticalsoftware/UserResource.java (hunk 11 above)
Comment (Leonorjp) [reply src/main/java/com/criticalsoftware/UserResource.java]: Done!

File: src/main/java/com/criticalsoftware/User.java (hunk 13 above)
Comment (Leonorjp) [reply src/main/java/com/criticalsoftware/User.java:9]: Done!

**Review by jpmoreira-training — APPROVED** [review]

**Pull request merged**